### DISCUSSION TITLE
Access programatic prerender initialPageContext in the onBeforeRender hook

### DISCUSSION THREAD

**Discussion by michaelhthomas** [vikejs/vike#574] [Discussion] [answered]
I'm trying to build something akin to the `graphql-apollo-vue` example, just using prerender instead of SSR. I'm not sure why, but it looks like the `initialPageContext` that is passed to `renderPage()` in the development server can be accessed just fine in `onBeforeRender`, but when doing prerender, `pageContext.apolloServer` is undefined.

Here is the prerender script I'm using for reference:

```js
const { ApolloClient, HttpLink, InMemoryCache } = require('@apollo/client')
const fetch = require('cross-fetch')
const { prerender } = require('vite-plugin-ssr/prerender')

function makeApolloClient() {
  const apolloClient = new ApolloClient({
    ssrMode: true,
    link: new HttpLink({
      uri: 'https://countries.trevorblades.com',
      fetch
    }),
    cache: new InMemoryCache()
  })
  return apolloClient
}

const apolloClient = makeApolloClient()
const pageContextInit = {
  apolloClient
}

prerender({
  pageContextInit
})
```

Is this expected behavior, and is there a way to work around it?

**Comment by brillout** [comment, marked as answer]
It was bug which just got fixed, see https://github.com/brillout/vite-plugin-ssr/issues/576.

In case your company is up for it: https://github.com/sponsors/brillout.
